src/Shopsys/ShopBundle/Model/Order/OrderDataMapper.php

```diff
@@ -0,0 +1,97 @@
+    {
+        $orderData = new OrderData();
+        $orderData->transport = $frontOrderData->transport;
+        $orderData->payment = $frontOrderData->payment;
+        $orderData->orderNumber = $frontOrderData->orderNumber;
+        $orderData->status = $frontOrderData->status;
+        $orderData->firstName = $frontOrderData->firstName;
+        $orderData->lastName = $frontOrderData->lastName;
+        $orderData->email = $frontOrderData->email;
+        $orderData->telephone = $frontOrderData->telephone;
+        $orderData->street = $frontOrderData->street;
+        $orderData->city = $frontOrderData->city;
+        $orderData->postcode = $frontOrderData->postcode;
+        $orderData->country = $frontOrderData->country;
+        $orderData->deliveryAddressSameAsBillingAddress = $frontOrderData->deliveryAddressSameAsBillingAddress;
+        $orderData->deliveryFirstName = $frontOrderData->deliveryFirstName;
+        $orderData->deliveryLastName = $frontOrderData->deliveryLastName;
+        $orderData->deliveryCompanyName = $frontOrderData->deliveryCompanyName;
+        $orderData->deliveryTelephone = $frontOrderData->deliveryTelephone;
+        $orderData->deliveryStreet = $frontOrderData->deliveryStreet;
+        $orderData->deliveryCity = $frontOrderData->deliveryCity;
+        $orderData->deliveryPostcode = $frontOrderData->deliveryPostcode;
+        $orderData->deliveryCountry = $frontOrderData->deliveryCountry;
+        $orderData->note = $frontOrderData->note;
+        $orderData->itemsWithoutTransportAndPayment = $frontOrderData->itemsWithoutTransportAndPayment;
+        $orderData->orderTransport = $frontOrderData->orderTransport;
+        $orderData->orderPayment = $frontOrderData->orderPayment;
+        $orderData->domainId = $frontOrderData->domainId;
+        $orderData->currency = $frontOrderData->currency;
+
+        $orderData->pickUpPlace = $orderData->transport !== null && $orderData->transport->getType() === Transport::TYPE_ZASILKOVNA ?
+                $frontOrderData->pickUpPlace : null;
+
+        $isPickUpPlaceTransport = $orderData->transport !== null && $orderData->transport->isPickUpPlaceType();
+        if ($isPickUpPlaceTransport && $orderData->pickUpPlace !== null) {
+            $this->fillPickUpPlaceDeliveryData($orderData, $frontOrderData);
+        }
+
+        if ($frontOrderData->companyCustomer) {
+            $orderData->companyName = $frontOrderData->companyName;
+            $orderData->companyNumber = $frontOrderData->companyNumber;
+            $orderData->companyTaxNumber = $frontOrderData->companyTaxNumber;
+        } else {
+            $orderData->companyName = null;
+            $orderData->companyNumber = null;
+            $orderData->companyTaxNumber = null;
+        }
+
+        return $orderData;
+    }
+
+    private function fillPickUpPlaceDeliveryData(OrderData $orderData, FrontOrderData $frontOrderData)
```

Comment:
I see, shipping address form is disabled and that is the reason why u need to fill the delivery address from chosen pickupPlace. Am I right? But if you set the shipping address form on readonly instead of disabled, this method will be unnecessary. Correct?

Reply:
after consultation, no change needed